kong-gateway-microshift.md

```diff
@@ -0,0 +1,691 @@
+enterprise:
+  enabled: true
+  license_secret: kong-enterprise-license
+  portal:
+    enabled: false
+  rbac:
+    enabled: false
+  smtp:
+    enabled: false
+admin:
+  enabled: false
+manager:
+  enabled: false
+portal:
+  enabled: false
+portalapi:
+  enabled: false  
+EOF
+helm install kong -n kong-dp kong/kong -f values.yaml
+```
+
+Wait for the data plane pod to be ready
+```bash
+oc wait --for=condition=ready pod -l app.kubernetes.io/component=app -n kong-dp --timeout=180s
+```
+Expose the Proxy Service
+```bash
+oc expose svc/kong-kong-proxy -n kong-dp --port=kong-proxy --hostname=kong-proxy.microshift.io
+
+oc create route passthrough kong-kong-proxy-tls --port=kong-proxy-tls --hostname=kong-proxy-tls.microshift.io --service=kong-kong-proxy -n kong-dp
+```
+
+Explain to our local node how to resolve requests to our routes:
+```bash
+export IP=$(oc get no -ojsonpath='{.items[*].status.addresses[?(@.type=="InternalIP")].address}')
+sudo sh -c  "echo $IP kong-proxy.microshift.io kong-proxy-tls.microshift.io >> /etc/hosts"
+```
+
+Validate `/etc/hosts` has been properly updated, expect to see kong-proxy, kong-proxy-tls
+```bash
+tail -1 /etc/hosts
+```
+output
+```text
+10.88.0.4 kong-proxy.microshift.io kong-proxy-tls.microshift.io
+```
+
+Check the Data Plane from the Control Plane to make sure the Data Plane is part of the cluster:
+```bash
+http `oc get route -n kong kong-kong-admin --template='{{ .spec.host }}'`/clustering/status
+```
+
+output
+```text
+HTTP/1.1 200 OK
+access-control-allow-origin: *
+cache-control: private
+content-length: 170
+content-type: application/json; charset=utf-8
+date: Wed, 01 Jun 2022 13:57:06 GMT
+deprecation: true
+server: kong/2.8.1
+set-cookie: 9da87f6e8821b5f9e46a0f05aee42078=5bd1a33699c17cb7a92c693484aa8ad8; path=/; HttpOnly
+x-kong-admin-latency: 5
+
+{
+    "b53487b3-752b-4ac5-8a64-f75e3592ca7c": {
+        "config_hash": "569818cb13aa3a90b1d72ca8225cd0cf",
+        "hostname": "kong-kong-77594db78-tkmz8",
+        "ip": "10.42.0.1",
+        "last_seen": 1654091809
+    }
+}
+```
+
+**NOTE** Notice that this part fails above 👆 (we should see the kong-dp-kong). This is an exploratory spike so lets just take note that it does not work as expected and move on. If you see a problem or have a solution- i'm all ears. This is the only thing that I cannot get to work, but the rest of the document works just fine. **This _could_ be a false positive.**
+
+
+Check the Data Plane Proxy to ensure that it is working:
+```bash
+http `oc get route -n kong-dp kong-kong-proxy --template='{{ .spec.host }}'`/
+```
+
+output
+```
+HTTP/1.1 404 Not Found
+cache-control: private
+content-length: 48
+content-type: application/json; charset=utf-8
+date: Wed, 01 Jun 2022 13:59:14 GMT
+server: kong/2.8.1
+set-cookie: 7439e381b0d6fc4efb69077feca119cd=479354089fd040bb701b8f071fefdb6c; path=/; HttpOnly
+x-kong-response-latency: 0
+
+{
+    "message": "no Route matched with those values"
+}
+```
+
+## Deploy Sample App
```

Comment:
I would deploy the app in a different ns, not the default because it might have different permissions/configuration. I have done it in a demo ns and it all works perfectly as well